sqlobject/src/main/java/org/jdbi/v3/sqlobject/SqlObjectFactory.java

```diff
@@ -127,6 +128,17 @@ private boolean looksLikeSqlObject(Class<?> extensionType) {
                 handlers.computeIfAbsent(method, m -> buildMethodHandler(sqlObjectType, m, registry, decorators));
             }
 
+            handlers.keySet().stream()
+                .filter(m -> !m.isSynthetic())
+                .collect(Collectors.groupingBy(m -> Arrays.asList(m.getName(), Arrays.asList(m.getParameterTypes()))))
+                .values()
```

Comment:
I wonder if this could be made more lazy, so we don't group everything needlessly when a single size>2 entry is enough. Then again, micro-optimization so let's not spend too much thought on it.

Reply:
This is only done once per class during initialization, so I don't think we should spend time micro-optimizing it.